### DISCUSSION TITLE
Why I cannot easily use built-in checks?

### DISCUSSION THREAD

**Discussion by MarcinOrlowski** [checkstyle/checkstyle#10248] [Q&A]
We use Checkstyle to guard our Java project. The approach is simple: built-in "Google Checks" and some suppressions. That's it. I only want to maintain my `suppressions.xml`. Nothing more. I have a InteliJ plugin, I have Gradle plugin. All fine. But when I try to achieve the same results using Checkstyle in shell, things are not that simple. I am simply not getting why. I am running Debian based distro so 

```bash
apt install checkstyle
```

and all good. BUT, to run it I need to give it a config file. I do not want to have my personal config here - I want to use **bundled**  `Google Checks` as my base, and add my suppression file on top of it. I do not want to keep copy of `google_checks.xml` (this can change across versions), not I want to do all the trickery with extracting it from JAR etc, etc. I am fine with whatever bundled Google checks is minus my suppression. Why cannot I simply do anything as straightforward as :

```bash
checkstyle --style=google-checks --suppression-file=/path/suppressions.xml [....]
```
Why is such usecase not supported even Checkstyle is at version 8 now? Do I miss something? What I do not understand?

**Comment by pbludov** [comment]
Related issues:
 #8418
#6942

**Comment by romani** [comment]
Thanks a lot for writing us.

You should use https://checkstyle.org/cmdline.html#Download_and_Run

`java -jar checkstyle-8.44-all.jar -c /google_checks.xml MyClass.java`

Path to config is resource in jar.

I hope you understand that google style is versioned by Google and by Checkstyle separately. 
Google style is just a config in checkstyle jar.

For suppressions please read https://checkstyle.org/google_style.html#Google_Suppressions 

Please read about different filters/suppression modules , whey are bit different and good for different goals https://checkstyle.org/config_filters.html

All is almost the already as you want but maybe names a bit different.
